pkg/cmd/adm/register_member.go

```diff
@@ -29,93 +38,177 @@ const (
+		" SpaceProvisionerConfig that you can modify with the required configuration options and apply to make the cluster available for space placement.", memberKubeconfig))
 
-	hostClusterConfig, err := configuration.LoadClusterConfig(ctx, configuration.HostName)
+	// construct the client to the host cluster and figure out the API endpoint of the member cluster. We will use this information later,
+	// but let's obtain it first so that we fail without leaving garbage if there is some kind of simple misconfiguration.
+	hostConfig, err := clientcmd.LoadFromFile(hostKubeconfig)
 	if err != nil {
 		return err
 	}
-	hostClusterClient, err := ctx.NewClient(hostClusterConfig.Token, hostClusterConfig.ServerAPI)
+	hostClientConfig, err := clientcmd.NewDefaultClientConfig(*hostConfig, nil).ClientConfig()
 	if err != nil {
 		return err
 	}
-
-	if err := disableAutomaticApproval(hostClusterConfig, hostClusterClient); err != nil {
+	hostClusterClient, err := ctx.NewClientFromRestConfig(hostClientConfig)
+	if err != nil {
 		return err
 	}
-
-	if err := runAddClusterScript(ctx, newCommand, configuration.Host, hostKubeconfig, memberKubeconfig); err != nil {
+	hostApiEndpoint, hostOperatorNamespace := getServerAPIEndpointAndNamespace(hostConfig)
+	if hostOperatorNamespace == "" {
+		hostClusterConfig, err := configuration.LoadClusterConfig(ctx, configuration.HostName)
+		if err != nil {
+			return fmt.Errorf("host operator namespace could not be determined and also failed to load the .ksctl.yaml: %w", err)
+		}
+		hostOperatorNamespace = hostClusterConfig.OperatorNamespace
+	}
+	memberConfig, err := clientcmd.LoadFromFile(memberKubeconfig)
+	if err != nil {
+		return err
+	}
+	memberApiEndpoint, memberOperatorNamespace := getServerAPIEndpointAndNamespace(memberConfig)
+	memberClientConfig, err := clientcmd.NewDefaultClientConfig(*memberConfig, nil).ClientConfig()
+	if err != nil {
 		return err
 	}
-	if err := runAddClusterScript(ctx, newCommand, configuration.Member, hostKubeconfig, memberKubeconfig); err != nil {
+	memberClusterClient, err := ctx.NewClientFromRestConfig(memberClientConfig)
+	if err != nil {
 		return err
 	}
 
-	warningMessage := "The automatic approval was disabled!\n Configure the new member cluster in ToolchainConfig and apply the changes to the cluster."
+	// as the first thing, let's check that the member has not been yet registered with the cluster.
+	hostsInMember := &toolchainv1alpha1.ToolchainClusterList{}
+	if err = memberClusterClient.List(ctx, hostsInMember, runtimeclient.InNamespace(memberOperatorNamespace)); err != nil {
+		return err
+	}
+	if len(hostsInMember.Items) > 0 {
+		return fmt.Errorf("the member cluster (%s) is already registered with some host cluster in namespace %s", memberApiEndpoint, memberOperatorNamespace)
+	}
```

Comment:
`add-cluster.sh` has an explicit `--multi-member` arg that enables you to re-apply the same combination of host + member repeatedly.

This command now tries to infer the value of `--multi-member` by detecting the next value, assuming you are adding a new member to a cluster where some members are already present.

So if you want to re-apply the same host + member, we need to add an equivalent to `--multi-member` to the register-member command.

Reply:
yes, let's add an equivalent flag to the --multi-member option :+1:

Reply:
I've put some new checks in to prevent damage to the cluster but enabled this one workflow (updating the existing toolchaincluster objects) with just a warning and required confirmation.